.github/workflows/integration.yml

```diff
@@ -110,376 +67,52 @@ jobs:
-        name: nats
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 30
-          max_attempts: 3
-          command: bash scripts/ci-int-e2e-test.sh int  nats
-
-      - run: docker image prune -af --filter=label!=vector-test-runner=true ; docker container prune -f
-
-      - if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.nginx == 'true' }}
-        name: nginx
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 30
-          max_attempts: 3
-          command: bash scripts/ci-int-e2e-test.sh int  nginx
-
-      - run: docker image prune -af --filter=label!=vector-test-runner=true ; docker container prune -f
-
-      - if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.opentelemetry == 'true' }}
-        name: opentelemetry
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 30
-          max_attempts: 3
-          command: bash scripts/ci-int-e2e-test.sh int  opentelemetry
-
-      - run: docker image prune -af --filter=label!=vector-test-runner=true ; docker container prune -f
-
-      - if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.postgres == 'true' }}
-        name: postgres
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 30
-          max_attempts: 3
-          command: bash scripts/ci-int-e2e-test.sh int  postgres
-
-      - run: docker image prune -af --filter=label!=vector-test-runner=true ; docker container prune -f
-
-      - if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.prometheus == 'true' }}
-        name: prometheus
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 30
-          max_attempts: 3
-          command: bash scripts/ci-int-e2e-test.sh int  prometheus
-
-      - run: docker image prune -af --filter=label!=vector-test-runner=true ; docker container prune -f
-
-      - if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.pulsar == 'true' }}
-        name: pulsar
-        uses: nick-fields/retry@v3
-        with:
-          timeout_minutes: 30
-          max_attempts: 3
-          command: bash scripts/ci-int-e2e-test.sh int  pulsar
-
-      - run: docker image prune -af --filter=label!=vector-test-runner=true ; docker container prune -f
-
-      - if: ${{ github.event_name == 'merge_group' || needs.changes.outputs.all-int == 'true' || needs.changes.outputs.redis == 'true' }}
-        name: redis
-        uses: nick-fields/retry@v3
+  integration-tests:
+    runs-on: ubuntu-20.04
+    needs:
+      - changes
+      - setup
+    strategy:
+      matrix:
+        service: [
+          "amqp", "appsignal", "aws", "axiom", "azure", "clickhouse", "databend", "datadog-agent",
```

Comment:
Might be worth adding a comment to this to point people to update the list in `changes.yml` if this one changes.